pkg/ebpf/bytecode/runtime/runtime_compilation_helpers.go

```diff
@@ -117,9 +117,10 @@ func compileToObjectFile(in io.Reader, outputDir, filename, inHash string, addit
 }
 
 func computeFlagsAndHash(additionalFlags []string) ([]string, string) {
-	flags := make([]string, len(defaultFlags)+len(additionalFlags))
-	copy(flags, defaultFlags)
-	copy(flags[len(defaultFlags):], additionalFlags)
+	flags := make([]string, len(defaultFlags)+len(additionalFlags)+1)
+	flags[0] = fmt.Sprintf("-D__TARGET_ARCH_%s", kernel.Arch())
+	copy(flags[1:], defaultFlags)
+	copy(flags[len(defaultFlags)+1:], additionalFlags)
```

Comment:
nit:
I don't think we care about allocations here, so I think this reads better and and it's more future-proof:
```go
var flags []string
flags = append(flags, defaultFlags...)
flags = append(flags, additionalFlags...) 
flags = append(flags, fmt.Sprintf("-D__TARGET_ARCH_%s", kernel.Arch())) 
```